static/js/lib/api.js

```diff
@@ -243,15 +244,23 @@ export function addCourseEnrollment(courseId: string) {
 }
 
 export function getCoupons(): Promise<Coupons> {
-  return fetchJSONWithCSRF('/api/v0/coupons/');
+  return fetchJSONWithCSRF('/api/v0/coupons/').then(coupons => {
+    if ( !coupons ) {
```

Comment:
It's necessary for the tests. Many of them stub out `fetchJSONWithCSRF` to return `undefined`:
```js
fetchJSONStub.returns(Promise.resolve());
```